javascript/frameworks/cap/lib/advanced_security/javascript/frameworks/cap/CDS.qll

```diff
@@ -739,5 +849,171 @@ class HandlerParameterData instanceof PropRead {
     )
   }
 
-  string toString() { result = this.(PropRead).toString() }
+  string toString() { result = super.toString() }
+}
+
+/**
+ * A call to a method capable of running a CQL query. This includes the following:
+ * - Generic query runners: `cds.run`, `cds.db.run`, `srv.run`
+ * - Shortcut to CQL's `READ`: `cds.read`, `cds.db.read`, `srv.read`
+ * - Shortcut to CQL's `CREATE`: `cds.create`, `cds.db.create`, `srv.create`
+ * - Shortcut to CQL's `INSERT`: `cds.insert`, `cds.db.insert`, `srv.insert`
+ * - Shortcut to CQL's `UPSERT`: `cds.upsert`, `cds.db.upsert`, `srv.upsert`
+ * - Shortcut to CQL's `UPDATE`: `cds.update`, `cds.db.update`, `srv.update`
+ * - Shortcut to CQL's `DELETE`: `cds.delete`, `cds.db.delete`, `srv.delete`
+ */
+abstract class CqlQueryRunnerCall extends MethodCallNode {
+  SourceNode base;
+  string methodName;
+
+  CqlQueryRunnerCall() {
+    this = base.getAMemberInvocation(methodName) and
+    (
+      /*
+       * 1. Method call on the CDS facade or the base database service,
+       * accessed as `cds.db`.
+       */
+
+      exists(CdsFacade cds | base = cds.asSource()) or
+      exists(CdsDb cdsDb | base = cdsDb) or
+      /*
+       * 2. Method call on a service instance object.
+       */
+
+      exists(ServiceInstance srv | base.getALocalSource() = srv) or
+      /*
+       * 3. Method call on a transaction object.
+       */
+
+      exists(CdsTransaction tx | base = tx)
+    )
+  }
+
+  /**
+   * Gets an argument to this runner call, including the subsequent builder functions
+   * called in a chained manner on this one.
+   */
+  abstract DataFlow::Node getAQueryParameter();
+}
+
+class CqlRunMethodCall extends CqlQueryRunnerCall {
+  CqlRunMethodCall() { this.getMethodName() = "run" }
+
+  override DataFlow::Node getAQueryParameter() { result = this.getArgument(0) }
+}
+
+/**
+ * A [CRUD-style call](https://cap.cloud.sap/docs/node.js/core-services#crud-style-api)
+ * that translates to running a CQL query internally.
+ */
+class CqlShortcutMethodCall extends CqlQueryRunnerCall {
+  CqlShortcutMethodCall() {
+    this.getMethodName() = ["read", "create", "update", "delete", "insert", "upsert"]
+  }
+
+  abstract override DataFlow::Node getAQueryParameter();
+}
+
+class CqlReadMethodCall extends CqlShortcutMethodCall {
+  CqlReadMethodCall() { this.getMethodName() = "read" }
+
+  override DataFlow::Node getAQueryParameter() {
+    result = this.getAChainedMethodCall(_).getAnArgument()
+  }
+}
+
+class CqlCreateMethodCall extends CqlShortcutMethodCall {
+  CqlCreateMethodCall() { this.getMethodName() = "create" }
+
+  override DataFlow::Node getAQueryParameter() {
+    exists(DataFlow::CallNode chainedMethodCall |
+      chainedMethodCall = this.getAChainedMethodCall(_)
+    |
+      result = chainedMethodCall.getAnArgument() or
+      result = chainedMethodCall.getAnArgument().(SourceNode).getAPropertyWrite().getRhs()
```

Comment:
I think we want to remove this case, otherwise it will still be a sink via `StringConcatParameterOfCqlShortcutMethodCall`.